cpp/common/src/codingstandards/cpp/exceptions/ExceptionFlow.qll

```diff
@@ -312,7 +313,21 @@ class ReThrowExprThrowingExpr extends ReThrowExpr, ThrowingExpr {
 
 /** An expression which calls a function which may throw an exception. */
 class FunctionCallThrowingExpr extends FunctionCall, ThrowingExpr {
-  override ExceptionType getAnExceptionType() { result = getAFunctionThrownType(getTarget(), _) }
+  override ExceptionType getAnExceptionType() {
+    exists(Function target |
+      target = getTarget() and
+      result = getAFunctionThrownType(target, _) and
+      // [expect.spec] states that throwing an exception type that is prohibited
+      // by the specification will result in the program terminating. We therefore
+      // do not propagate such exceptions to the call sites for the function.
```

Comment:
```suggestion
      // [expect.spec] states that throwing an exception type that is prohibited
      // by the specification will result in the program terminating, unless
      // a custom `unexpected_handler` is registered that throws an exception type
      // which is compatible with the specification.
      // We assume a default `std::unexpected` handler and therefore
      // do not propagate such exceptions to the call sites for the function.
      //
```

Reply:
Added, although I reworded it slightly for clarity.